nav2_motion_primitives/src/spin.cpp

```diff
@@ -146,4 +146,10 @@ nav2_tasks::TaskStatus Spin::controlledSpin()
   return TaskStatus::RUNNING;
 }
 
+bool Spin::pathIsClear()
```

Comment:
See https://github.com/ros-planning/navigation/blob/kinetic-devel/rotate_recovery/src/rotate_recovery.cpp#L105

No need to reinvent the wheel :)

Reply:
Yeah, the code in `controlledSpin()` is based on ROS1 `rotate_recovery` controller. I haven't added a service for checking for free space within the robot's own [footprint](https://github.com/ros-planning/navigation/blob/kinetic-devel/rotate_recovery/src/rotate_recovery.cpp#L122).